File: javascript/leetcode/747. Largest Number At Least Twice of Others.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var dominantIndex = function (nums) {
  let ans = -1;

  let mx = Math.max(...nums);

  for (let i = 0; i < nums.length; ++i) {
    if (nums[i] !== mx && nums[i] * 2 > mx) return -1;

    if (mx === nums[i]) ans = i;
  }

  return ans;
};

File: c++/leetcode/747. Largest Number At Least Twice of Others.cpp
class Solution
{
public:
  int dominantIndex(vector<int> &nums)
  {
    int ans = -1;

    auto mx = *max_element(nums.begin(), nums.end());

    for (int i = 0; i < nums.size(); ++i)
    {
      if (nums[i] != mx && (nums[i] * 2) > mx)
        return -1;

      if (mx == nums[i])
        ans = i;
    }

    return ans;
  }
};